Tests/test_GenBank_unittest.py

```diff
@@ -688,6 +689,145 @@ def test_000_write_invalid_but_parsed_locus_line(self):
+        l_r = self.gb_to_l_r("GenBank/NC_005816.gb", do_features=False)
+        # number of records, should be 1
+        self.assertEqual(len(l_r), 1)
+        self.assertEqual(l_r[0].id, 'NC_005816.1')
+        self.assertEqual(l_r[0].name, 'NC_005816')
+        self.assertEqual(l_r[0].description, 'Yersinia pestis biovar '
+                                             'Microtus str. 91001 plasmid '
+                                             'pPCP1, complete sequence')
+        self.assertEqual(len(l_r[0].features), 0)
+
+        # Test parse records on NC_005816, do_features True
+        l_r = self.gb_to_l_r("GenBank/NC_005816.gb", do_features=True)
+        # number of records, should be 1
+        self.assertEqual(len(l_r), 1)
+        self.assertEqual(l_r[0].id, 'NC_005816.1')
+        self.assertEqual(l_r[0].name, 'NC_005816')
+        self.assertEqual(l_r[0].description, 'Yersinia pestis biovar '
+                                             'Microtus str. 91001 plasmid '
+                                             'pPCP1, complete sequence')
+        self.assertEqual(len(l_r[0].features), 41)
+
+        # Test parse records on "GenBank/NC_000932.gb", do_features False
+        l_r = self.gb_to_l_r("GenBank/NC_000932.gb", do_features=False)
+        # number of records, should be 1
+        self.assertEqual(len(l_r), 1)
+        self.assertEqual(l_r[0].id, 'NC_000932.1')
+        self.assertEqual(l_r[0].name, 'NC_000932')
+        self.assertEqual(l_r[0].description, 'Arabidopsis thaliana chloroplast, '
+                                             'complete genome')
+        self.assertEqual(len(l_r[0].features), 0)
+
+        # Test parse records on NC_000932, do_features True
+        l_r = self.gb_to_l_r("GenBank/NC_000932.gb", do_features=True)
+        # number of records, should be 1
+        self.assertEqual(len(l_r), 1)
+        self.assertEqual(l_r[0].id, 'NC_000932.1')
+        self.assertEqual(l_r[0].name, 'NC_000932')
+        self.assertEqual(l_r[0].description, 'Arabidopsis thaliana chloroplast, '
+                                             'complete genome')
+        self.assertEqual(len(l_r[0].features), 259)
+
+    def test_embl_cds_interaction(self):
+        """Test EMBL CDS interaction, parse CDS features on embl files."""
+
+        # Test parse CDS features on embl_file
+        handle_embl7046 = open("EMBL/AE017046.embl", "r")
+        embl_s = Scanner.EmblScanner()
+        l_cds_f = list(embl_s.parse_cds_features(handle_embl7046))
+        handle_embl7046.close()
+
+        # number of records, should be 10
+        self.assertEqual(len(l_cds_f), 10)
+        # Seq ID
+        self.assertEqual(l_cds_f[0].id, 'AAS58758.1')
+        self.assertEqual(l_cds_f[0].description, 'putative transposase')
+
+    def test_embl_record_interaction(self):
+        """Test EMBL Record interaction on embl files."""
+
+        #  Test parse records on embl_file
+        handle_embl7046 = open("EMBL/AE017046.embl", "r")
```

Comment:
Again, please use a context manager.

Reply:
Using context manager for all now; all good, I think is ready now.